src/api.js

```diff
@@ -0,0 +1,182 @@
+let debug = require('debug')('events:api');
+let APIBuilder = require('taskcluster-lib-api');
+let taskcluster = require('taskcluster-client');
+let uuid = require('uuid');
+let _ = require('lodash');
+
+let builder = new APIBuilder({
+  title: 'AMQP Messages API Documentation',
+  description: [
+    'This service is responsible for making pulse messages accessible',
+    'from browsers and cli. There are API endpoints to',
+    'bind / unbind to an exchange and pause / resume listening from a queue',
+  ].join('\n'),
+  projectName: 'taskcluster-evnets',
+  serviceName: 'events',
+  version: 'v1',
+  context: ['connection'],
+});
+
+// Returns JSON.parse(bindings) if everything goes well
+//   {"bindings" : [ 
+//     {"exchange" :  "a/b/c", "routingKey" : "a.b.c"},
+//     {"exchange" :  "x/y/z", "routingKey" : "x.y.z"},
+//   ]};
+var parseAndValidateBindings = function(bindings) {
+  return new Promise((resolve, reject) => {
+    try {
+      let json_bindings = JSON.parse(bindings);
+      if (String(Object.keys(json_bindings)) !== String(['bindings'])) {
+        throw new Error('The json query should have only one key i.e. `bindings`.');
+      }  
+
+      // Reduce json_bindings to an array of exchanges.
+      json_bindings = json_bindings.bindings;
+      if (!Array.isArray(json_bindings)) {
+        throw new Error('Bindings must be an array of {exchange, routingKey}');
+      }
+      json_bindings.map(binding => {
```

Comment:
`map` returns a new array. If you don't care about the return value, I would use `forEach`.

Reply:
Yes, right. 
I don't know why `_.forEach` didn't occur to me . Thanks